### ISSUE TITLE
Support for bracket notation in v-model directive

### ISSUE THREAD

**Issue by dgerber** [vuejs/vue#279]
Is there a way to support the following use case?

Binding

``` html
<div v-repeat="items">
  <input v-model="index[id]">
</div>
```

to

``` js
{  items: [ {id: 'key1', ..}, {id: 'key2', ..}, index: { key1: 'somevalue', .. } ]
```

One limitation is that all properties have to be defined on `index` beforehand (before binding, and for all values of `id`). It seems #246 would solve this part.

A second problem is that the current v-model directive does not support assigning to an expression with brackets. Syncing from model to view works fine after #251, but not the other way around.

**Comment by davidkhess** [comment]
I'm running into this same issue. I've trying to build a form based on data in one part of the model and I need to set bind the input values to another part of the model (based on this data).

I would be happy to entertain work arounds? So far, I've had to do watches on the both ends of the model in order to make it work.

I think what would work is not completely dynamic two way binding but a one-time evaluation of mustache in reactive expressions. Sort of like how a dynamic literal is evaluated once on a directive with no update function.

It's not as powerful as completely dynamic two way binding but it would solve this problem well enough.

**Comment by yyx990803** [comment]
This should already work in 0.12 as long as you've initialized the data properly, can you provide a fiddle where it's not working as intended?

**Comment by davidkhess** [comment]
I think this demonstrates it:

https://jsfiddle.net/davidkhess/0qqLnsyr/1/

**Comment by davidkhess** [comment]
@yyx990803, any chance we can reopen this? Or, I'll be happy to open a new issue.

**Comment by marcnewport** [comment]
+1

**Comment by NachoPer** [comment]
+

**Comment by NachoPer** [comment]
+1
